Source/engine/displacement.hpp

```diff
@@ -200,14 +161,14 @@ struct Displacement {
 	 *
 	 * This will return a displacement of the form (-1.0 to 1.0, -0.5 to 0.5), to get a full tile offset you can multiply by 16
 	 */
-	Displacement worldToNormalScreen() const
+	[[nodiscard]] Displacement worldToNormalScreen() const
 	{
 		// Most transformations between world and screen space take shortcuts when scaling to simplify the math. This
 		//  routine is typically used with missiles where we want a normal vector that can be multiplied with a target
 		//  velocity (given in pixels). We could normalize the vector first but then we'd need to scale it during
 		//  rotation from world to screen space. To save performing unnecessary divisions we rotate first without
 		//  correcting the scaling. This gives a vector in elevation projection aligned with screen space.
-		Displacement rotated { (deltaY - deltaX), -(deltaY + deltaX) };
+		Displacement rotated { static_cast<int>(deltaY) - static_cast<int>(deltaX), -(static_cast<int>(deltaY) + static_cast<int>(deltaX)) };

```

Comment:
I think you can avoid the cast here by allowing this to be the current type, it's only the return value from `DisplacementOf<T>::normalized()` that needs an explicit type.
```suggestion
		DisplacementOf<DeltaT> rotated { deltaY - deltaX, -(deltaY + deltaX) };
```

though I guess it needs a static assert like other operations expecting DeltaT to be signed, I think having the assert is preferable personally but happy to go either way.

Reply:
I was too slow 😂 . Followup maybe?

Reply:
I've added the casts because of the compiler warnings but feel free to experiment and send a follow-up now that this is merged!